instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/SpanFactory.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.instrumentation.awssdk.v2_2;
+
+import io.opentelemetry.instrumentation.api.tracer.BaseTracer;
+import io.opentelemetry.trace.DefaultSpan;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Span.Kind;
+import io.opentelemetry.trace.Tracer;
+import io.opentelemetry.trace.TracingContextUtils;
+import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.core.interceptor.SdkExecutionAttribute;
+
+public class SpanFactory {
```

Comment:
Why this was extracted from `AwsSdkClientTracer`?

Reply:
Separation of concerns / testability. Although I have not added a unit test, I'd be much easier now.

Reply:
All other Tracers do create spans. If you extract that from them, what will remain? :)

Reply:
Pure tracing code ;) I don't have any other explanation for this refactoring apart from object oriented design principles :)

Reply:
I sort of agree that extracting this class doesn't seem to have huge bang for buck and makes this instrumentation significantly different from many of our other tracers. But it's ok too. Let's make sure all the new classes are package private instead of public

Reply:
I still think I disagree with this refactoring atm. We are still in the process of migrating from old Decorators to new Tracers and of unifying their usage. Introducing one more pattern of structuring and using tracers will only impede that unification.

Sorry to be a party pooper, but I would ask to move it back :)

Reply:
No problem with that.